lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -114,3 +114,52 @@ function gutenberg_override_default_rest_server() {
 	return 'Gutenberg_REST_Server';
 }
 add_filter( 'wp_rest_server_class', 'gutenberg_override_default_rest_server', 1 );
+
+/**
+ * Adds user avatar URLs to the REST API response for WordPress version 6.7 and below.
+ *
+ * This function is used to add avatar URLs for the author of a post in the REST API response.
+ * It checks if the function 'add_user_avatar_urls_in_rest_response_6_7' does not already exist,
+ * and if not, defines the function.
+ *
+ * @param object $response The REST API response object.
+ * @return object The modified REST API response object with added avatar URLs.
+ */
+if ( ! function_exists( 'add_user_avatar_urls_in_rest_response_6_7' ) && gutenberg_is_experiment_enabled( 'gutenberg-block-comment' ) ) {
+	function add_user_avatar_urls_in_rest_response_6_7( $response ) {
+
+		if ( $response->data['author'] ) {
+			$response->data['author_avatar_urls'] = array(
+				'default' => get_avatar_url( $response->data['author'] ),
+				'48'      => add_query_arg( 's', 48, get_avatar_url( $response->data['author'], array( 'size' => 48 ) ) ),
+				'96'      => add_query_arg( 's', 96, get_avatar_url( $response->data['author'], array( 'size' => 96 ) ) ),
+			);
+		}
+
+		return $response;
+	}
+	add_filter( 'rest_prepare_comment', 'add_user_avatar_urls_in_rest_response_6_7', 10, 1 );
```

Comment:
As [also mentioned](https://github.com/WordPress/gutenberg/pull/60622#discussion_r1750319899) in #60622:

This will only work when retrieving comments via the REST API. If we want to retrieve them on the backend, it won't work. 

I'm trying to understand why `get_comment` will set the avatar URL to `null` (as mentioned [here](https://github.com/WordPress/gutenberg/pull/60622#discussion_r1747081608)). A few possibilities come to mind:

- is it because the `show_avatars` option is not enabled? 
- is it because the `author_avatar_urls` field is not enabled when performing the REST API request?

Can you speak more about where this assignment to `null` happens?

In any case, to achieve what you're doing it should be enough to just use the `rest_avatar_sizes` filter and the `rest_get_avatar_urls()` function would build the avatar URLs properly. If we're just doing it for the REST API response, we're doing it in an inconsistent way which is likely to introduce unexpected bugs and missing data in unexpected places.

Reply:
The issue has been resolved. We found that the REST API was returning a null avatar URL because it only supported the default '**comment**' type while we were using a **custom type**. 

To fix this, we applied the `get_avatar_comment_types` filter to include our custom comment type, 'block_comment'. This allowed us to remove the `rest_prepare_comment `filter, and the issue is now resolved.

Reply:
Great, thanks 🙌